### PR TITLE
pre-commit hook that checks for mock.patch invocations without autospec

### PR DESCRIPTION
**Pull request by mjksmith**
Example:

```
(general_itests)matts@some-hostname:~/paasta (master) $ pre-commit run --all-files
Trim Trailing Whitespace...........................................................................................................................................Passed
Fix End of Files...................................................................................................................................................Passed
autopep8 wrapper...................................................................................................................................................Passed
Check docstring is first...........................................................................................................................................Passed
Check JSON.........................................................................................................................................................Passed
Check Yaml.........................................................................................................................................................Passed
Debug Statements (Python)..........................................................................................................................................Passed
Tests should end in _test.py.......................................................................................................................................Passed
Fix requirements.txt...............................................................................................................................................Passed
Flake8.............................................................................................................................................................Passed
Reorder python imports.............................................................................................................................................Failed
hookid: reorder-python-imports

Files were modified by this hook.

mock.patch enforce autospec........................................................................................................................................Failed
hookid: patch-enforce-autospec

tests/cli/test_cmds_check.py:42: Found a mock without an autospec!
tests/cli/test_cmds_check.py:43: Found a mock without an autospec!
tests/cli/test_cmds_check.py:44: Found a mock without an autospec!
tests/cli/test_cmds_check.py:45: Found a mock without an autospec!
...............
tests/test_marathon_serviceinit.py:958: Found a mock without an autospec!
tests/test_marathon_serviceinit.py:968: Found a mock without an autospec!
You probably meant to specify 'autospec=True' in these tests.
If you really can't use autospec, please specify 'autospec=False'

```

### CODE REVIEW

**Comment by mjksmith** [comment]
The tests that are failing are from the new pre-commit hook and are resolved in #770

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -8,18 +8,26 @@
...
         language_version: python2.7
 -   repo: https://github.com/asottile/reorder_python_imports.git
-    sha: 3d86483455ab5bd06cc1069fdd5ac57be5463f10
+    sha: b022734351abe44d0b05a71a4fa1175287c59b49
     hooks:
     -   id: reorder-python-imports
         language_version: python2.7
+-   repo: local
+    hooks:
+    -   id: patch-enforce-autospec
```
Comment (ealter) [inline .pre-commit-config.yaml]: You should pin the python version too.

Comment (mjksmith) [inline .pre-commit-config.yaml]: Does that work even though I'm using the `script` language instead of the `python` language?

**Comment by solarkennedy** [comment]
ship

**Pull request merged**